src/winrpath.h

```diff
@@ -0,0 +1,249 @@
+#pragma once
+
+#include <stdlib.h>
+#include <string>
+#include <stdio.h>
+#include <vector>
+#include <map>
+#include <iostream>
+#include <fstream>
+#include <windows.h>
+#include <winnt.h>
+#include <strsafe.h>
+
+#include "utils.h"
+#include "execute.h"
+
+// Structs for holding coff/PE data
+// winnt headers define a number of
+// PE/COFF components as structs
+// Here we define structs to hold and
+// organize those system defined structs
+// and define our own in the cases winnt
+// misses some
+
+#pragma pack(push, r1, 1)
+
+/**
+ * @brief
+ */
+typedef struct long_import_member {
+    PIMAGE_FILE_HEADER pfile_h;
+    PIMAGE_SECTION_HEADER * pp_sections;
+    char ** section_data;
+    PIMAGE_SYMBOL * symbol_table;
+    char * string_table;
+    DWORD size_of_string_table;
+    long long string_table_offset;
+    ~long_import_member() {
+        for (int i=0; i<this->pfile_h->NumberOfSymbols; ++i) {
+            delete *(this->symbol_table+i);
+        }
+        for (int i=0; i< this->pfile_h->NumberOfSections; ++i) {
+            delete *(this->section_data+i);
+            delete *(this->pp_sections+i);
+        }
+        delete this->symbol_table;
+        delete this->section_data;
+        delete this->pp_sections;
+        if(this->string_table) {
+            delete this->string_table;
+        }
+    }
+} long_import_member;
+
+/**
+ * @brief
+ */
+typedef struct short_import_member {
+    char * short_name;
+    char * short_dll;
+    IMPORT_OBJECT_HEADER *im_h;
+} short_import_member;
+
+/**
+ * 
+ */
+typedef struct first_linker_member {
+    DWORD symbols;
+    PDWORD offsets;
+    char * strings;
+} first_linker_member;
+
+/**
+ * 
+ */
+typedef struct second_linker_member {
+    DWORD members;
+    PDWORD offsets;
+    DWORD symbols;
+    PWORD indicies;
+    char * strings;
+} second_linker_member;
+
+/**
+ * 
+ */
+typedef struct longnames_member {
+    char * names_field;
+} longnames_member;
+
+
+/**
+ * @brief
+ */
+typedef struct coff_member {
+    char * data;
+    bool is_short;
+    short_import_member *short_member;
+    long_import_member *long_member;
+    first_linker_member *first_link;
+    second_linker_member *second_link;
+    coff_member() {
+        this->is_short = false;
+        this->first_link = NULL;
+        this->second_link = NULL;
+        this->short_member = NULL;
+        this->long_member = NULL;
+    }
+    ~coff_member () {
+        if (this->is_short) {
+            delete short_member;
+        }
+        else {
+            delete long_member;
+        }
```

Comment:
(minor) my understanding is that there's no issue with `delete`ing null members.

Reply:
Good point!